cunumeric/deferred.py

```diff
@@ -624,13 +624,25 @@ def get_item(self, key):
                     self.dtype,
                     inputs=[self],
                 )
-                copy = self.context.create_copy()
+                if index_array.base.kind == Future:
+                    # TODO replace with .copy() call when copy() uses COPY task
+                    # instead of the UNARY_OP.COPY
+                    task = self.context.create_task(CuNumericOpCode.COPY)
+                    task.add_output(result.base)
+                    task.add_input(store)
+                    task.add_input(index_array.base)
+                    task.add_scalar_arg(True, bool)  # source indirect
+                    task.add_alignment(result.base, index_array.base)
+                    task.add_broadcast(store)
```

Comment:
I don't like very much the idea of broadcasting input array, but I am not sure how to partition it when I have an indirection

Reply:
@magnatelee In this particular example I use `copy` task only when indirect field is a `Future`.  So the question is: how would I convert `Future` to the `Store`?

Reply:
It's done in two steps. First, you need to create a store of volume 1 that is backed by a region field. You could do it by passing `optimize_scalar=False` when creating the store: https://github.com/nv-legate/cunumeric/blob/branch-22.05/cunumeric/runtime.py#L415-L419 Then, you copy a future-backed store to that new store. I guess you might want to add a runtime method for doing it, as it's a common utility.

Reply:
I see. How do I decide that store is a big store? Should I just always convert `Future` to `Store` and then use CopyLauncher in this case?

Reply:
If I do that I wouldn't need to have a separate COPY task at all

Reply:
Right. if any of the three stores can be a regular store, I think you should be converting the others to regular stores and using the Legate copy. (I thought the case this PR is addressing is when all three stores are backed by futures.)

Reply:
Then I will close this PR and open another one that takes proposed approach.  Thank you!